cns/fakes/imdsclientfake.go

```diff
@@ -41,3 +45,15 @@ func (c *WireserverClientFake) GetInterfaces(ctx context.Context) (*wireserver.G
 		},
 	}, nil
 }
+
+func NewMockIMDSClient() *MockIMDSClient {
+	return &MockIMDSClient{}
+}
+
+func (m *MockIMDSClient) GetVMUniqueID(ctx context.Context) (string, error) {
+	if ctx.Value(SimulateError) != nil {
+		return "", imds.ErrUnexpectedStatusCode
+	}
```

Comment:
An easier way to do this is to have the struct be an empty husk of funcs that you can override at will in tests:

```go
type MockIMDSClient struct {
       GetVMUniqueIDF func(context.Context) (string, error)
}

func (m *MockIMDSClient) GetVMUniqueID(ctx context.Context) (string, error) {
       return m.GetVMUniqueIDF(ctx)
}

// ===
// in test:

func TestThing(t *testing.T) {
    // snip
    m := &MockIMDSClient{
              GetVMUniqueIDF: func(ctx context.Context) (string, error) {
                     return "", errors.New("boom")
              }
    }
}

Reply:
I am trying to perform an end to end via API handlers. IMDSClient is mostly an internal and not really exposed to the test. But I like the approach and can use it in other places.